lib/int3.h

```diff
@@ -92,6 +92,31 @@ class int3
 		return false;
 	}
 
+	enum EDistanceFormula
+	{
+		DIST_2D = 0,
+		DIST_MANHATTAN, // patrol distance
+		DIST_CHEBYSHEV, // ambient sound distance
+		DIST_2DSQ
+	};
+
+	ui32 dist(const int3 & o, EDistanceFormula formula) const

```

Comment:
Is this formula going to be used elsewhere? If in RMG or AI, I'd already implement it as a template + inlines to avoid additional overhead.
But probably it's not necessary to reuse it, anyway.

Reply:
I don't think it's will be heavily used since almost everything only need ```dist2d``` / ```dist2dSQ```. Manhattan distance used for patrol only and chebyshev only for ambient sounds that are not really performance critical, but it's still nice to explicitly specify what formula is used especially considering that we might need some other weird formulas in future.